### PR TITLE
Cleaned up challenges and jobs

### PR DESCRIPTION
**Pull request by russjones**
Cleaned up challenges and jobs.

* Create README for all interview challenges
* Renamed "cloud" to "sre" and removed out of date leveling guide
* Renamed "frontend" challenge to "fullstack" to reflect scope
* Removed unused "product-design" challenge
* Removed empty frontend jobs posting
* Removed trailing whitespace from challenges

### CODE REVIEW

**Comment by russjones** [comment]
Let's not merge this until https://github.com/gravitational/careers/pull/68 is merged.

**Review by r0mant — APPROVED** [review]

**Review by jimbishopp — APPROVED** [review]

File: challenges/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Teleport Interview Challenges
+
+### Systems Engineering
+
+Teleport has two systems engineering challenges.
+
+* [Job Worker](systems/challenge-1.md): A challenge that demonstrates understanding of
+  Linux internals.
+* [Prototype TCP Load Balancer](systems/challenge-2.md): A challenge that demonstrates
```
Comment (zmb3) [inline challenges/README.md:9]:
```suggestion
* [TCP Load Balancer](systems/challenge-2.md): A challenge that demonstrates
```

There both prototypes, technically. Let's keep it simple.

File: challenges/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+We also maintain a [systems engineering challenge for interns](systems-intern/challenge.md).
+
+### Fullstack Engineering
+
+Teleport has one fullstack engineering challenge.
+
+* [Directory Browser](fullstack/challenge.md): A challenge that demonstrates
+  understanding of frontend and backend engineering.
+
+### Site Reliability Engineer
```
Comment (zmb3) [inline challenges/README.md:21]:
```suggestion
### Site Reliability Engineering
```

In order to be consistent with the other section headers.

File: challenges/fullstack/challenge.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
+# Frontend Application Developer
```
Comment (zmb3) [inline challenges/fullstack/challenge.md:1]:
```suggestion
# Fullstack Application Developer
```

**Review by zmb3 — APPROVED** [review]

**Pull request merged**